### DISCUSSION TITLE
Designing for Non-coders

### DISCUSSION THREAD

**Discussion by kerryhill** [MasaCMS/MasaCMS#232] [General]
I'm building a Masa CMS site for a client with no coding skills and I have a few questions. What is the best way for them to be able to update their header/footer? For example, they may want a logo in the header or want to add social media links in the footer. 
 
Here is the functionality WordPress provides:
![image](https://github.com/MasaCMS/MasaCMS/assets/35034737/9003fc4e-1add-4a4e-97a9-0e2812a61500)

Also, in WP there are lots of performed content blocks available that are helpful for content managers and I was wondering how everyone else is addressing this in Masa.

Thanks

**Comment by Tigerlady** [comment]
Kerry,  I have done a few workarounds for this. 

First option is started in site settings, Display Regions.  I have one site site up with 10 display regions that I use on various page templates used throughout the site.  
![image](https://github.com/MasaCMS/MasaCMS/assets/1850313/e294c29c-2343-4043-8356-43df33a28db9)

Then in my code, I will lay out that these are editable areas depending on the template being used for example #$.dspObjects(4)# or
`<div class="feed-wrapper2">
		#$.dspObjects(5)#
		</div>`

Sometimes I will also create a component that can be edited by the user that will not directly effect the design but they can change information as needed for example I might create a footer logo component and hard code the component in my code surrounded by what ever CSS is needed.  The example below is code I have in the footer 

`	<div class="single-widget fnav widget">
							#$.dspObject('component','footer-nav')#
						</div>		
            </div>
							
            <div class="col-lg-3 col-md-6 col-12">
						<div class="single-widget fnav widget">
            #$.dspObject('component','Certifications')#
            
            </div>`

Also if you want to add things like linked in or facebook or even phone numbers or emails you can add these as extend attributes to the site itself adding to the class extensions. Here is that it looks like on the site information.

![image](https://github.com/MasaCMS/MasaCMS/assets/1850313/71fd8279-99bb-4d9c-9d26-d693c2c624dd)
![image](https://github.com/MasaCMS/MasaCMS/assets/1850313/b5e83479-9a09-4b87-9520-ee8f24afe737)

I hope this helps.

**Comment by kerryhill** [comment]
Thanks!! Lots to unpack here but that last part where you added the extended attribute tab to the site settings. I'm very interested in how you did that.  That looks amazing!

**Reply by Tigerlady** [reply]
Go to Site, Class Extensions, if you do not have one for site, click add Class Extension, From the Base Type select Site from the dropdown menu.

![image](https://github.com/MasaCMS/MasaCMS/assets/1850313/cb593562-5b68-4894-80e5-116a81bfdae5)

Then from that, add the attribute set
![image](https://github.com/MasaCMS/MasaCMS/assets/1850313/47194dc2-e3c4-4ba7-bf3b-ce55533b8cba)

Give it a name and have it go to Extended Attributes Tab.  You can create your own custom UI, but this is the easiest. 

![image](https://github.com/MasaCMS/MasaCMS/assets/1850313/3d5c7121-d30e-4571-9962-e20a2af0af8f)

Then add a new Attribute.

![image](https://github.com/MasaCMS/MasaCMS/assets/1850313/56a08a90-262d-4131-9b8a-72d1177439af)

Then select what you want.  For me, it is a TextBox where the link would go.

![image](https://github.com/MasaCMS/MasaCMS/assets/1850313/8225a298-0b66-428d-a4b1-95f4537d24be)

You can add several Attribute sets.

once you have those, you or your admin users can fill in the information as needed to display on the page. 
I usually wrap mine in an if statement similar to this.
`<cfif len($.siteConfig('TagLine'))><p>#$.siteConfig('TagLine')#`
Replace TagLine with whatever you named the attribute.  ie LinkedInURL

Hope this helps.

**Reply by guustnieuwenhuis** [reply]
Extended attributes on site is also the way we implement functionality like this. Thank you @Tigerlady to help out @kerryhill.

A third option, when using the Layout Manager for the complete site design, is to create custom modules. For this I suggest reading through the [Mura documentation](https://docs.murasoftware.com/v7-1/mura-developers/mura-rendering/#section-mura-modules-display-objects). We don't have a dedicated documentation for this in Masa CMS, yet.

**Comment by Tigerlady** [comment]
Kerry,  Just curious did my examples help?

**Reply by kerryhill** [reply]
Yes, thank you very much for the detailed examples. I don't know why I was having so much trouble figuring it out from the docs.
